src/rpc/RPCHelpers.cpp

```diff
@@ -312,7 +313,7 @@ getDeliveredAmount(
         // then its absence indicates that the amount delivered is listed in the
         // Amount field. DeliveredAmount went live January 24, 2014.
         // 446000000 is in Feb 2014, well after DeliveredAmount went live
-        if (ledgerSequence >= 4594095)
+        if (ledgerSequence >= 4594095 || date > 446000000)
         {
             return txn->getFieldAmount(ripple::sfAmount);
```

Comment:
Is it _guaranteed_ that in this case `sfAmount` will be `delivered_amount`?

If a transaction is a partial payment, will it _always_ have `delivered_amount` in the metadata?

Reply:
According to the comment (which I copied from rippled), delivered amount will be in the metadata for partial payments after this ledger. The metadata is checked prior to this, at the beginning of the function.